src/main.cpp

```diff
@@ -1295,6 +1304,7 @@ void readBitmapData(WiFiClient &client)
   bool has_multicolors = false;
   bool grayscale = false;
 
+  bool read_error;
```

Comment:
Move the initialisation into group above like `bool read_error = false;`
On the other hand, can't you use `valid` instead?